pkg/cmd/pr/checkout/checkout.go

```diff
@@ -268,3 +295,71 @@ func executeCmds(client *git.Client, credentialPattern git.CredentialPattern, cm
 	}
 	return nil
 }
+
+func resolvePR(httpClient *http.Client, baseRepo ghrepo.Interface, prompter shared.Prompter, pullRequestSelector string, isInteractive bool, pullRequestFinder shared.PRFinder, cs *iostreams.ColorScheme) (*api.PullRequest, error) {
+	// When non-interactive
+	if pullRequestSelector != "" {
+		pr, _, err := pullRequestFinder.Find(shared.FindOptions{
+			Selector: pullRequestSelector,
+			Fields: []string{
+				"number",
+				"headRefName",
+				"headRepository",
+				"headRepositoryOwner",
+				"isCrossRepository",
+				"maintainerCanModify",
+			},
+		})
+		if err != nil {
+			return nil, err
+		}
+		return pr, nil
+	}
+	if !isInteractive {
+		return nil, cmdutil.FlagErrorf("pull request number, URL, or branch required when not running interactively")
+	}
+	// When interactive
+	listResult, err := list.ListPullRequests(httpClient, baseRepo, shared.FilterOptions{Entity: "pr", State: "open", Fields: []string{
+		"number",
+		"title",
+		"state",
+		"url",
+		"isDraft",
+		"createdAt",
+
+		"headRefName",
+		"headRepository",
+		"headRepositoryOwner",
+		"isCrossRepository",
+		"maintainerCanModify",
+	}}, 10)
+	if err != nil {
+		return nil, err
+	}
+
+	pr, err := promptForPR(prompter, cs, *listResult)
+
+	return pr, err
+}
+
+func promptForPR(prompter shared.Prompter, cs *iostreams.ColorScheme, jobs api.PullRequestAndTotalCount) (*api.PullRequest, error) {
+	candidates := []string{}
+	for _, pr := range jobs.PullRequests {
+		candidates = append(candidates, text.Truncate(120, fmt.Sprintf("%s %s (%s)",
+			shared.PRNumberWithColor(cs, pr),
+			text.RemoveExcessiveWhitespace(pr.Title),
+			cs.Gray(pr.HeadLabel()),
+		)))
```

Comment:
```suggestion
		candidates = append(candidates, text.Truncate(120, fmt.Sprintf("%s %s [%s]",
			shared.PRNumberWithColor(cs, pr),
			text.RemoveExcessiveWhitespace(pr.Title),
			cs.Gray(pr.HeadLabel()),
		)))
```

💭 I think this should be in square brackets to align with [Primer guidelines on displaying branches](https://primer.style/native/cli/components#branches)

Reply:
gotcha! I will update it per your suggestion 
Initially, I was following the design of the prompt `gh run view`
<img width="696" alt="Screenshot 2024-12-06 at 5 59 55 pm" src="https://github.com/user-attachments/assets/2f954f02-01ee-4148-9278-bbf03d21d4ac">

maybe we can update them to make it consistent as well

Reply:
Good catch! 

I opened https://github.com/cli/cli/issues/10038. Assuming the team agrees, it seems like something that would be a good first issue for someone 😁

Reply:
thanks @BagToad! 🙌